pkg/build/controller/controller.go

```diff
@@ -190,22 +201,23 @@ func (bc *BuildController) nextBuildPhase(build *buildapi.Build) error {
 
 	if _, err := bc.PodManager.CreatePod(build.Namespace, podSpec); err != nil {
 		if errors.IsAlreadyExists(err) {
-			bc.Recorder.Eventf(build, kapi.EventTypeWarning, "failedCreate", "Pod already exists: %s/%s", podSpec.Namespace, podSpec.Name)
+			bc.Recorder.Eventf(build, kapi.EventTypeWarning, "FailedCreate", "Pod already exists: %s/%s", podSpec.Namespace, podSpec.Name)
 			glog.V(4).Infof("Build pod already existed: %#v", podSpec)
 			return nil
 		}
 		// Log an event if the pod is not created (most likely due to quota denial).
 		bc.Recorder.Eventf(build, kapi.EventTypeWarning, "FailedCreate", "Error creating: %v", err)
 		build.Status.Reason = buildapi.StatusReasonCannotCreateBuildPod
+		build.Status.Message = StatusMessageCannotCreateBuildPod
 		return fmt.Errorf("failed to create build pod: %v", err)
 	}
 	setBuildPodNameAnnotation(build, podSpec.Name)
 	glog.V(4).Infof("Created pod for build: %#v", podSpec)
 
 	// Set the build phase, which will be persisted.
 	build.Status.Phase = buildapi.BuildPhasePending
-	build.Status.Reason = ""
-	build.Status.Message = ""
+	// build.Status.Reason = ""
+	// build.Status.Message = ""
```

Comment:
there should be no harm in setting it to empty here, right?  you're only worried about resetting it to empty when you've indicated a failure reason, but if the phase is pending, you haven't set a failure reason.